### PR TITLE
moved to more es6 for stuff

### PR DESCRIPTION
**Pull request by jhorwit2**

### CODE REVIEW

File: src/APIActionCreator.es6.js [hunk 1]
```diff
@@ -98,7 +110,7 @@ export default class APIActionCreator extends ActionCreator {
 		);
 
 		invariant(
-			typeof description.method === 'string',
+			typeof method === 'string' || method,
```
Comment (mako-taco) [inline src/APIActionCreator.es6.js]: doesnt this mean if method is ANYTHING it will pass the check?

File: src/APIActionCreator.es6.js [hunk 2]
```diff
@@ -90,7 +102,7 @@ export default class APIActionCreator extends ActionCreator {
 		);
 
 		invariant(
-			typeof description.route === 'string' || description.route,
+			typeof route === 'string' || route,
```
Comment (mako-taco) [inline src/APIActionCreator.es6.js]: drop the `|| route` I have no idea what that is for

File: src/APIActionCreator.es6.js (hunk 1 above)
Comment (jhorwit2) [inline src/APIActionCreator.es6.js]: We actually do this check often for strings. 

`typeof description.route === 'string' || description.route,`

For example, which is 3 lines above this check.

**Comment by jhorwit2** [comment]
@mako-taco fixed those 2 invariant checks.

File: src/debug.es6.js [hunk 3]
```diff
@@ -120,28 +117,31 @@ class FluxDebugger {
...
 	 */
 	registerAction (actionCreator, action) {
-		var existingEntry;
+		let {source, type} = action;
+		let existingEntry;
 
-		if(!this.registeredActions[action.source]) {
-			this.registeredActions[action.source] = {};
-			this.registeredSources[action.source] = actionCreator;
+		if(!this.registeredActions[source]) {
```
Comment (mako-taco) [inline src/debug.es6.js]: lol noob add a space

**Pull request merged**